section_vector_index.go

```diff
@@ -211,6 +211,73 @@ func (v *vectorIndexOpaque) mergeAndWriteVectorIndexes(fieldID int, sbs []*Segme
 		vecIndexes = append(vecIndexes, index)
 	}
 
+	if len(vecToDocID) > 10000 {
```

Comment:
Should perhaps make this configurable?

If you chose 10000 only because faiss advertises the flat indexes perform well up until 10000 - I'm thinking if we should do some testing on our side to decide what a better value for this could be.